File: TASK 1/Question 5.cpp
#include <iostream>
#include <iomanip>
using namespace std;

int main() {
    int decimalNumber;

    
    cout << "Enter a decimal number: ";
    cin >> decimalNumber;

    
    cout << "Hexadecimal: 0x" << hex << decimalNumber << endl;
    cout << "Octal: 0" << oct << decimalNumber << endl;
    cout << "Binary: ";
    
    int binary[32];
    int index = 0;

    
    while (decimalNumber > 0) {
        binary[index] = decimalNumber % 2;
        decimalNumber /= 2;
        index++;
    }

    
    for (int i = index - 1; i >= 0; i--) {
        cout << binary[i];
    }
    cout << endl;

    return 0;
}

File: TASK 2/Question 5.py
def decimal_to_binary(decimal_num):
    if decimal_num <= 0:
        return "0"
    elif decimal_num == 1:
        return "1"
    else:
        
        return decimal_to_binary(decimal_num // 2) + str(decimal_num % 2)

decimal_num = int(input("Enter a decimal number: "))

binary_num = decimal_to_binary(decimal_num)
print("Binary representation:", binary_num)
